test/ng/rootScopeSpec.js

```diff
@@ -2316,6 +2316,99 @@ describe('Scope', function() {
         }));
 
 
+        // See issue https://github.com/angular/angular.js/issues/16135
+        it('should deallocate the listener array entry', inject(function($rootScope) {
+          var remove1 = $rootScope.$on('abc', noop);
+          $rootScope.$on('abc', noop);
+
+          expect($rootScope.$$listeners['abc'].length).toBe(2);
+          expect(0 in $rootScope.$$listeners['abc']).toBe(true);
+
+          remove1();
+
+          expect($rootScope.$$listeners['abc'].length).toBe(2);
+          expect(0 in $rootScope.$$listeners['abc']).toBe(false);
+        }));
+
+
+        it('should call next listener when removing current', inject(function($rootScope) {
```

Comment:
I just realized there are two similar tests to this one where the current listener gets removed for [$emit](https://github.com/angular/angular.js/blob/d0d829899c4bcdf4b322d45245d7b811cd4cb855/test/ng/rootScopeSpec.js#L2438) and [$broadcast](https://github.com/angular/angular.js/blob/d0d829899c4bcdf4b322d45245d7b811cd4cb855/test/ng/rootScopeSpec.js#L2472) (https://github.com/angular/angular.js/commit/e6966e05f508d1d2633b9ff327fea912b12555ac) under `describe('$emit'`. Maybe those two should move here to replace this one? Or just keep this extra one?